airbyte-workers/src/main/java/io/airbyte/workers/internal/AirbyteMessageTracker.java

```diff
@@ -145,10 +154,16 @@ private void handleSourceEmittedRecord(final AirbyteRecordMessage recordMessage)
    * correctly.
    */
   private void handleSourceEmittedState(final AirbyteStateMessage stateMessage) {
-    updateMaxAndMeanSecondsToReceiveStateMessage(DateTime.now());
+    final DateTime timeEmittedStateMessage = DateTime.now();
+    updateMaxAndMeanSecondsToReceiveStateMessage(timeEmittedStateMessage);
     sourceOutputState.set(new State().withState(stateMessage.getData()));
     totalSourceEmittedStateMessages.incrementAndGet();
     final int stateHash = getStateHashCode(stateMessage);
+
+    if (AirbyteStateType.LEGACY != stateMessage.getType()) {
```

Comment:
Why do we avoid the tracking if the state is legacy?

Reply:
we need to track the emit -> commit timing data per stream, and with legacy we can't track per stream state messages so it requires a separate code path, and since it's deprecated anyway we decided against supporting it.

Reply:
now I'm actually getting confused about how state is stored for global vs per stream vs legacy though. in airbyte_protocol.yaml it looks like connections with GLOBAL state should look at the data field (the same as legacy) but I thought this was updated?

Reply:
Global state should eventually be reading the `AirbyteStateMessage::global` attribute not the data. I thought reading data was part of our backward compatible implementation.